src/qmapshack/gis/search/CSearchLineEdit.h

```diff
@@ -0,0 +1,67 @@
+/**********************************************************************************************
+    Copyright (C) 2019 Henri Hornburg hrnbg@t-online.de
+
+    This program is free software: you can redistribute it and/or modify
+    it under the terms of the GNU General Public License as published by
+    the Free Software Foundation, either version 3 of the License, or
+    (at your option) any later version.
+
+    This program is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU General Public License for more details.
+
+    You should have received a copy of the GNU General Public License
+    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+**********************************************************************************************/
+#ifndef CSEARCHLINEEDIT_H
+#define CSEARCHLINEEDIT_H
+
+#include "CSearchExplanationDialog.h"
+#include "gis/search/CSearch.h"
+#include <canvas/CCanvas.h>
+#include <QLineEdit>
+#include <QMenu>
```

Comment:
Please check if all these headers are need and/or can be replaced by forward declarations.